### PR TITLE
feat: set up farm and product fields

### PR DESCRIPTION
**Pull request by ces-huy-dang**
create fields for farm and product to design farm homepage and product details page

CU-865d32kjk

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/Code-Engine-Studio/ces-semantic-release/blob/main/README.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by nguyenvuces — APPROVED** [review]

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 1]
```diff
@@ -83,11 +105,11 @@ public function getImagesUrlFromArray($store_images_field) {
    * @param array $products_array
    *   An product array.
    *
-   * @return products
+   * @return productsList
    *   Return array of product to display on UI
    */
   public function getProductsInfomation($products_array) {
-    $products = [];
+    $productsList = [];
```
Comment (nguyenvuces) [inline web/modules/custom/farm_page/src/Services/FarmPageService.php:112]: Consistently use Camel case or Snake case

**Review by nguyenvuces — CHANGES_REQUESTED** [review]

File: config/core.entity_view_display.commerce_product.default.default.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ dependencies:
     - field.field.commerce_product.default.field_category
     - field.field.commerce_product.default.field_images
     - field.field.commerce_product.default.field_isfeature
+    - field.field.commerce_product.default.field_taglines
```
Comment (ces-dantran) [inline config/core.entity_view_display.commerce_product.default.default.yml:11]: field_tag

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 3]
```diff
@@ -29,6 +30,27 @@ public function getStoreIdByUserId($uid) {
     return $store_id;
   }
 
+  /**
+   * Get Taxonomy term of an Object from taxonomy_id .
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Services/FarmPageService.php:34]: Common doesn't make sense.

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 4]
```diff
@@ -83,11 +105,11 @@ public function getImagesUrlFromArray($store_images_field) {
...
    *   An product array.
    *
-   * @return products
+   * @return products_list
    *   Return array of product to display on UI
    */
   public function getProductsInfomation($products_array) {
-    $products = [];
+    $products_list = [];
     foreach ($products_array as $product_in_array) {
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Services/FarmPageService.php:113]: foreach ($products as $product)

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 5]
```diff
@@ -108,11 +130,23 @@ public function getProductsInfomation($products_array) {
       $product['product_description'] = $product_description[0]['value'];
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Services/FarmPageService.php:130]: $product['description']

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 6]
```diff
@@ -108,11 +130,23 @@ public function getProductsInfomation($products_array) {
...
       // Get Category.
       $product_category_id = $product_in_array->get('field_category')->getValue();
-      $product_category_name = Term::load($product_category_id[0]['target_id']);
-      $product['product_category'] = $product_category_name->getName();
-      array_push($products, $product);
+      $product_category = $this->getTaxonomyTerm($product_category_id);
+      $product['product_category'] = $product_category;
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Services/FarmPageService.php:134]: $product['category'] = $this->getTaxonomyTerm($term->tid);

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 7]
```diff
@@ -108,11 +130,23 @@ public function getProductsInfomation($products_array) {
...
+      // Get Variations.
+      $product_variation_id_list = $product_in_array->get('variations')->getValue();
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Services/FarmPageService.php:136]: $product_variations

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 8]
```diff
@@ -108,11 +130,23 @@ public function getProductsInfomation($products_array) {
...
+      if ($product_variation_id_list != NULL) {
+        $product_price_list = [];
+        foreach ($product_variation_id_list as $product_variation_id) {
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Services/FarmPageService.php:139]: foreach ($variations as $variation)

File: web/modules/custom/farm_page/templates/farm-homepage.html.twig [hunk 9]
```diff
@@ -1,8 +1,35 @@
-{% if is_exist == true %}
-{{ store.field_description.value }}
-	{% for image in store_images %}
-	<img src={{image}}>
-	{% endfor %}
+{% if data.is_exist == TRUE %}
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/templates/farm-homepage.html.twig:1]: code format.

**Review by ces-dantran — CHANGES_REQUESTED** [review]

File: web/modules/custom/farm_page/src/Services/FarmPageService.php (hunk 3 above)
Comment (ces-huy-dang) [reply web/modules/custom/farm_page/src/Services/FarmPageService.php]: fixed to "Get Taxonomy term by taxonomy_id"

File: web/modules/custom/farm_page/src/Services/FarmPageService.php (hunk 5 above)
Comment (ces-huy-dang) [reply web/modules/custom/farm_page/src/Services/FarmPageService.php]: fixed

File: web/modules/custom/farm_page/src/Services/FarmPageService.php (hunk 6 above)
Comment (ces-huy-dang) [reply web/modules/custom/farm_page/src/Services/FarmPageService.php]: fixed

File: web/modules/custom/farm_page/src/Services/FarmPageService.php (hunk 7 above)
Comment (ces-huy-dang) [reply web/modules/custom/farm_page/src/Services/FarmPageService.php]: fixed

File: web/modules/custom/farm_page/src/Services/FarmPageService.php (hunk 8 above)
Comment (ces-huy-dang) [reply web/modules/custom/farm_page/src/Services/FarmPageService.php]: fixed

File: web/modules/custom/farm_page/templates/farm-homepage.html.twig (hunk 9 above)
Comment (ces-huy-dang) [reply web/modules/custom/farm_page/templates/farm-homepage.html.twig:1]: formatted

File: config/core.entity_view_display.commerce_product.default.default.yml (hunk 2 above)
Comment (ces-huy-dang) [reply config/core.entity_view_display.commerce_product.default.default.yml]: fixed to field_tags

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 10]
```diff
@@ -29,6 +30,27 @@ public function getStoreIdByUserId($uid) {
...
+  /**
+   * Get Taxonomy term by taxonomy_id .
+   *
+   * @param mixed $taxonomy_id_list
+   *   user id from URL.
+   *
+   * @return terms_list
+   *   Return Store id to access to Store infomation
+   */
+  public function getTaxonomyTerm($taxonomy_id_list) {
```
Comment (ces-dantran) [inline web/modules/custom/farm_page/src/Services/FarmPageService.php:42]: getTaxonomyTerm($term) {

File: web/modules/custom/farm_page/templates/farm-homepage.html.twig (hunk 9 above)
Comment (ces-dantran) [reply web/modules/custom/farm_page/templates/farm-homepage.html.twig:1]: it's not formatted, should be two spaces.

**Review by ces-dantran — CHANGES_REQUESTED** [review]

File: web/modules/custom/farm_page/src/Services/FarmPageService.php (hunk 3 above)
Comment (ces-dantran) [reply web/modules/custom/farm_page/src/Services/FarmPageService.php]: Change it to: Get term by tid.

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 11]
```diff
@@ -29,6 +30,27 @@ public function getStoreIdByUserId($uid) {
     return $store_id;
   }
 
+  /**
+   * Get Taxonomy term by taxonomy_id .
+   *
+   * @param mixed $taxonomy_id_list
+   *   user id from URL.
```
[the remark this answers is not in the document]
Comment (ces-huy-dang) [reply web/modules/custom/farm_page/src/Services/FarmPageService.php]: fixed

File: web/modules/custom/farm_page/src/Services/FarmPageService.php (hunk 3 above)
Comment (ces-huy-dang) [reply web/modules/custom/farm_page/src/Services/FarmPageService.php]: Changed

File: web/modules/custom/farm_page/src/Services/FarmPageService.php [hunk 12]
```diff
@@ -29,6 +30,27 @@ public function getStoreIdByUserId($uid) {
     return $store_id;
   }
 
+  /**
+   * Get term by tid .
+   *
+   * @param mixed $terms
+   *   terms from Objects.
```
Comment (ces-khoado) [inline web/modules/custom/farm_page/src/Services/FarmPageService.php:37]: Capitalize first word.

Comment (ces-huy-dang) [reply web/modules/custom/farm_page/src/Services/FarmPageService.php]: Fixed

**Review by ces-dantran — APPROVED** [review]

**Review by nguyenvuces — APPROVED** [review]

**Pull request merged**